Allow redirect from non-https origins

tevosouza opened the pull request.

volkflo commented:
Great, I will try this probably the next days 😊

From: T3v0us <notifications@github.com>
Sent: Monday, 4 January 2021 15:21
To: apiman/apiman <apiman@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [apiman/apiman] Allow redirect from non-https origins (#947)

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/apiman/apiman/pull/947

Commit Summary

  *   Allow redirect from non-https origins

File Changes

  *   M distro/data/src/main/resources/data/apiman-realm-for-keycloak.json<https://github.com/apiman/apiman/pull/947/files#diff-2c737efabd451289d125065641bbc12112a7fe57428f77ac5bd6ff945daa1a4a> (13)
  *   M distro/wildfly/src/main/resources/overlay/standalone/data/h2/keycloak.mv.db<https://github.com/apiman/apiman/pull/947/files#diff-f86f0d2ea33fdbfa86526d8a79e5ceab660206db1556f3fcf9265af30d00f40f> (0)

Patch Links:

  *   https://github.com/apiman/apiman/pull/947.patch
  *   https://github.com/apiman/apiman/pull/947.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/apiman/apiman/pull/947>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADFW7H6MEFTA72AQ3SWMUX3SYHFGTANCNFSM4VTCRLMQ>.

msavy commented on distro/data/src/main/resources/data/apiman-realm-for-keycloak.json:
```diff
@@ -108,17 +108,19 @@
             ],
             "secret" : "password",
-            "webOrigins" : [ "*" ]
+            "webOrigins" : [ "+" ]
```
Is this a reference to this? https://lists.jboss.org/pipermail/keycloak-user/2017-October/012008.html

tevosouza replied:
Could be! heh
I made this change for security purposes, if we use '*' any weborigin is allowed to consume resources as a client, when I set to the '+' only valid redirects (set on the client configuration) is allowed as weborigins and it prevents unauthorized access and solves some CORS problem too, but CORS problem is not the case :).

msavy commented:
Is the addition of the H2 `keycloak.mv.db` necessary? Interested to hear thoughts on this. 🤔

tevosouza commented:
Hi @msavy,
When I did my tests to change redirect I changed only the JSON file, but it doesn't make any effect and then I put apiman up and changed config on keycloak and realized that H2 database is entirely deployed on the build and the JSON realm configuration file is not imported on stack startup.
Maybe we can change it via CLI to process realm configuration at keycloak startup but I think that we need to split apiman and master realms because keycloak base clients are exported too and it maybe causes initialization problems (i don't tested it yet...)

msavy commented:
Yes, it might just be a side-effect/autogeneration and possibly can be excluded safely. Give it a try when you have time 👍

volkflo commented:
@msavy I made the last changes on the `keycloak.mv.db` and forced everyone to use https. This file must be included because the quickstart does not automatically import the `apiman-realm-for-keycloak.json`. (Not sure if this is has worked before?).

At the moment this is more or less a template if you want to start with a standalone deployment.

msavy commented:
Aha, that sounds familiar, so should be ok! 

Thank you
